src/snark/src/algebra/fields/tests/test_bigint.cpp

```diff
@@ -24,84 +26,72 @@ void test_bigint()
     bigint<1> b1 = bigint<1>(b1_decimal);
     bigint<2> b2 = bigint<2>(b2_decimal);
 
-    assert(b0.as_ulong() == 0ul);
-    assert(b0.is_zero());
-    assert(b1.as_ulong() == 76749407ul);
-    assert(!(b1.is_zero()));
-    assert(b2.as_ulong() == 15747124762497195938ul);
-    assert(!(b2.is_zero()));
-    assert(b0 != b1);
-    assert(!(b0 == b1));
-
-    assert(b2.max_bits() == 128);
-    assert(b2.num_bits() == 99);
+    EXPECT_EQ(b0.as_ulong(), 0ul);
+    EXPECT_TRUE(b0.is_zero());
+    EXPECT_EQ(b1.as_ulong(), 76749407ul);
+    EXPECT_FALSE(b1.is_zero());
+    EXPECT_EQ(b2.as_ulong(), 15747124762497195938ul);
+    EXPECT_FALSE(b2.is_zero());
+    EXPECT_NE(b0, b1);
+    EXPECT_FALSE(b0 == b1);
+
+    EXPECT_EQ(b2.max_bits(), 128);
+    EXPECT_EQ(b2.num_bits(), 99);
     for (size_t i = 0; i < 128; i++) {
-        assert(b2.test_bit(i) == (b2_binary[127-i] == '1'));
+        EXPECT_EQ(b2.test_bit(i), (b2_binary[127-i] == '1'));
     }
 
     bigint<3> b3 = b2 * b1;
 
-    assert(b3 == bigint<3>(b3_decimal));
-    assert(!(b3.is_zero()));
+    EXPECT_EQ(b3, bigint<3>(b3_decimal));
+    EXPECT_FALSE(b3.is_zero());
 
     bigint<3> b3a { b3 };
-    assert(b3a == bigint<3>(b3_decimal));
-    assert(b3a == b3);
-    assert(!(b3a.is_zero()));
+    EXPECT_EQ(b3a, bigint<3>(b3_decimal));
+    EXPECT_EQ(b3a, b3);
+    EXPECT_FALSE(b3a.is_zero());
 
     mpz_t m3;
     mpz_init(m3);
     b3.to_mpz(m3);
     bigint<3> b3b { m3 };
-    assert(b3b == b3);
+    EXPECT_EQ(b3b, b3);
 
     bigint<2> quotient;
     bigint<2> remainder;
     bigint<3>::div_qr(quotient, remainder, b3, b2);
-    assert(quotient.num_bits() < GMP_NUMB_BITS);
-    assert(quotient.as_ulong() == b1.as_ulong());
+    EXPECT_LT(quotient.num_bits(), GMP_NUMB_BITS);
+    EXPECT_EQ(quotient.as_ulong(), b1.as_ulong());
     bigint<1> b1inc = bigint<1>("76749408");
     bigint<1> b1a = quotient.shorten(b1inc, "test");
-    assert(b1a == b1);
-    assert(remainder.is_zero());
+    EXPECT_EQ(b1a, b1);
+    EXPECT_TRUE(remainder.is_zero());
     remainder.limit(b2, "test");
 
-    try {
-        (void)(quotient.shorten(b1, "test"));
-        assert(false);
-    } catch (std::domain_error) {}
-    try {
-        remainder.limit(remainder, "test");
-        assert(false);
-    } catch (std::domain_error) {}
+    EXPECT_THROW((void)(quotient.shorten(b1, "test")), std::domain_error);
```

Comment:
I'm confused if you need the `assert(false)` in the new version

Reply:
No; the `assert(false)` in the original was so that the test would fail if the function call did not throw an error (note that the `catch` block is empty, so the test would continue). Google Test has a single macro for that.

Reply:
Ohh I see now